doc/interop-test-descriptions.md

```diff
@@ -679,6 +679,38 @@ Client asserts:
 by the auth library. The client can optionally check the username matches the
 email address in the key file.
 
+### google_default_credentials
+
+Similar to the other auth tests, this test should only be run against prod
+servers. Different from some of the other auth tests however, this test
+may be also run from outside of GCP.
+
+This test verifies unary calls succeed when the client uses
+GoogleDefaultCredentials. The path to a service account key file in the
+GOOGLE_APPLICATION_CREDENTIALS environment variable may or may not be
+provided by the test runner. For example, the test runner might set
+this environment when outside of GCP but keep it unset when on GCP.
+
+The test uses `--default_service_account` with GCE service account email.
+
+Server features:
+* [UnaryCall][]
+* [Echo Authenticated Username][]
+
+Procedure:
+ 1. Client configures the channel to use GoogleDefaultCredentials
```

Comment:
Should this include ALTS support, or no? Since many languages had google default credentials but they were just CallCredentials, it would probably be good to specify this to be either channel credentials or call credentials. (or note explicitly whether auto-selection of ALTS is expected)

Reply:
This is an odd point, I think related to @jtattermusch comment above about call creds vs. GoogleDefaultCreds.

Added a note in the description: my initial idea for this test was only to test the "GoogleDefaultCredentials" APIs that have ALTS auto-selection. In the future, we could add a test service that `run_interop_tests.py` could run against which would <i>only</i> works with clients that have such an API. In that case, if we were also using this test to run clients with the "call creds only" version of this API, I'd be worried that the test would start to get confusing.

That said, open to adding the call creds only version of the API to this test. Another idea is to have one more test, called e.g. "application default credential", which won't ever test ALTS auto-selection